File: 09.Greed贪心/leetcode_1846_maximumElementAfterDecrementingAndRearranging.py
from typing import List

class Solution:
    def maximumElementAfterDecrementingAndRearranging(self, arr: List[int]) -> int:
        arr.sort()
        cur = 1
        for i in arr[1:]:
            cur = min(cur+1, i)
        return cur

if __name__ == "__main__":
    s = Solution()
    arr = [2,2,1,2,1]
    print(s.maximumElementAfterDecrementingAndRearranging(arr))

File: 09.Greed贪心/leetcode_1846_maximumElementAfterDecrementingAndRearranging.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int maximumElementAfterDecrementingAndRearranging(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        int cur = 1;
        for (int i=1; i<arr.size(); i++){
            cur = min(cur+1, arr[i]);
        }
        return cur;
    }
};

int main(){
    Solution s;
    vector<int> arr = {2,2,1,2,1};
    cout << s.maximumElementAfterDecrementingAndRearranging(arr) << endl;
}
